src/GAP/iso_oscar_gap.jl

```diff
@@ -55,12 +55,16 @@ function _iso_oscar_gap(RO::Union{Nemo.zzModRing, Nemo.ZZModRing})
    return MapFromFunc(RO, RG, f, finv)
 end
 
-# Assume that `FO` and `FG` are finite fields of the same order
-# in Oscar and GAP, respectively.
-function _iso_oscar_gap_field_finite_functions(FO::Union{Nemo.fpField, Nemo.FpField}, FG::GAP.GapObj)
+_ffe_to_int(a::FqFieldElem) = Nemo._coeff(a, 0)
+_ffe_to_int(a::Union{fpFieldElem,FpFieldElem}) = lift(a)
+
+function _make_prime_field_functions(FO, FG)
    e = GAPWrap.One(FG)
 
-   f(x) = GAP.Obj(lift(x))*e
+   f = function(x)
+     y = GAP.Obj(_ffe_to_int(x))::GapInt
+     return (y*e)::GAP.FFE
```

Comment:
If `FO` is `Nemo.FpField(p)` with large enough `p` then `e` and `y*e` will not have the type `GAP.FFE`.
(They will be in the GAP filter `IsFFE` but not in `ÌsInternalRep`.)